### PR TITLE
Streamff extractor

### PR DESCRIPTION
**Pull request by cntrl-s**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

implement #1359

### CODE REVIEW

File: yt_dlp/extractor/streamff.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..utils import int_or_none, parse_iso8601
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:3]: just a suggestion (it's not needed per say tbh)
But i personally would have kept it same as all other extractors 
```suggestion
# coding: utf-8
from __future__ import unicode_literals

from .common import InfoExtractor
from ..utils import int_or_none, parse_iso8601
```

File: yt_dlp/extractor/streamff.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class StreamFFIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?streamff\.com/v/(?P<id>[a-zA-Z0-9]+)'
+    _API_URL = 'https://streamff.com/api'
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:8]: only used once so remove this from here

File: yt_dlp/extractor/streamff.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    _TESTS = [{
+        'url': 'https://streamff.com/v/55cc94',
+        'md5': '8745a67bb5e5c570738efe7983826370',
+        'info_dict': {
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:13]: add `view_count`

File: yt_dlp/extractor/streamff.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        """"
+        {
+            'views': 94,
+            'uploaded': True,
+            'publicURl': '55cc94',
+            'date': '2021-10-20T21:17:23.887Z',
+            'name': '55cc94',
+            'videoLink': '/uploads/55cc94.mp4'
+        }
+        """
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:35]: remove this

File: yt_dlp/extractor/streamff.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        json_data = self._download_json(self._API_URL + api_path, video_id)
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:36]:
```suggestion
        json_data = self._download_json(f'https://streamff.com/api/videos/{video_id}', video_id)
```

File: yt_dlp/extractor/streamff.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        video_url = f'https://streamff.com/{json_data["videoLink"]}'
+        title = json_data.get('name') or video_id
+
+        return {
+            'id': video_id,
+            'title': title,
+            'url': video_url,
+            'view_count': int_or_none(json_data.get('views')),
+            'timestamp': parse_iso8601(json_data.get('date'))
+        }
```
Comment (Ashish0804) [inline yt_dlp/extractor/streamff.py:46]:
```suggestion
        return {
            'id': video_id,
            'title': json_data.get('name') or video_id,
            'url': 'https://streamff.com/%s' % json_data['videoLink'],
            'view_count': int_or_none(json_data.get('views')),
            'timestamp': parse_iso8601(json_data.get('date')),
        }
```

File: yt_dlp/extractor/streamff.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/streamff.py:3]: No, if anything we should be removing the future imports from new extractors

**Comment by cntrl-s** [comment]
hi @Ashish0804 i was busy, i see you have made the changes yourself

**Pull request merged**